### ISSUE TITLE
Reshaping Last convolutional Layer

### ISSUE THREAD

**Issue by imraghs** [pradyu1993/segnet#2]
Hi,

You are reshaping the output of last convolutional layer to  ( 12 , 360 * 480 ) and permuting it to ( 360 * 480 , 12 )
so output shape after permutation will be ( None , 360 * 480 , 12 ) where None is the batch size as mentioned in ipython notebook.

On this  ( None , 360 * 480 , 12 ) , a 3D tensor, you are applying softmax, which according to keras softmax defination will be consider as ( nb_samples , nb_timesteps , nb_channels ).

My question is what is this nb_timesteps?? which is 360*480 for here. As per my understanding in semantic segmentation each pixel is sample so output should be ( None * 360 * 480 , 12 ) , but here we don't know value of None in prior, so can you give me explanation of how to make sense of your implementation, as I am unable to understand this nb_timesteps part.

**Comment by pradyu1993** [comment]
Hey @imraghs, softmax expected shape is either (nb_samples, nb_timesteps, nb_dims) or (nb_samples, nb_dims). Since softmax is applied on the last dimension I permuted the tensor so that input of softmax layer is 360*480,12
